qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -742,39 +658,164 @@ def risefall_sigma_ratio(self):
         return (self.duration - self.width) / (2.0 * self.sigma)
 
 
+class GaussianSquareEcho(SymbolicPulse):
+    """An echoed Gaussian square pulse.
+    Exactly one of the ``risefall_sigma_ratio`` and ``width`` parameters has to be specified.
```

Comment:
We need description of the pulse shape (mathematical, or in some sentence) and it would be great if you have some reference paper. Perhaps user may want to know how this differs from `GaussianSquare` and when they can use this pulse. 

Since this description about rise fall parameter is written in `GaussianSquare`, you can use [`.. seealso` directive](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html?highlight=See%20also#directive-seealso) to document link to that class and avoid duplication.